pkg/apiserver/proxy.go

```diff
@@ -286,6 +286,11 @@ func dialURL(url *url.URL, transport http.RoundTripper) (net.Conn, error) {
 			return nil, err
 		}
 
+		// Return if we were configured to skip validation
```

Comment:
upgrading connections to a TLS backend wasn't honoring tlsConfig.InsecureSkipVerify flag. Added tests for honoring the insecure flag to skip hostname validation and honoring a root ca bundle on the transport to verify the connection (here and for the generic rest proxy)